### ISSUE TITLE
Week view incorrectly displays events when a day contains overlapping events

### ISSUE THREAD

**Issue by TomJanse** [SimformSolutionsPvtLtd/flutter_calendar_view#130]
When two events in the week view are overlapping they become half as wide to fit over each other. This is perfectly fine. The problem is that when this happens at any time in that day **all** other events that day adapt the same half-size measure even if they are not overlapping with anything.

I would expect only the overlapping events to become small and all other events to remain the same.

My week view looks as follows:
```dart
WeekView(
      key: _weekViewStateKey,
      heightPerMinute: heightPerMinute,
      onEventTap: _onEventTap(),
      weekPageHeaderBuilder: (weekStartDate, weekEndDate) {
        return DayPageHeader(
          date: weekStartDate,
          dateStringBuilder: (date, {DateTime? secondaryDate}) {
            return DateFormat.yMMMd(Platform.localeName).format(weekStartDate) +
                ' - ' +
                DateFormat.yMMMd(Platform.localeName).format(weekEndDate);
          },
          onNextDay: () {
            _weekViewStateKey.currentState!.nextPage();
          },
          onPreviousDay: () {
            _weekViewStateKey.currentState!.previousPage();
          },
          backgroundColor: CupertinoDynamicColor.resolve(
              CupertinoColors.secondarySystemBackground, context),
        );
      },
    );
``` 

Visual of the week view with the issue happening:
<img width="345" alt="Screenshot 2022-11-04 at 09 53 12" src="https://user-images.githubusercontent.com/78813883/199932306-8d82b168-a656-4963-b8f3-967302ccb85a.png">

In the image, if we take a look at Thursday for example, the events from ~12:30-2:30 overlaps with ~2:30-4:30. It may look a little weird but the end and start of them are at the exact same moment so that makes sense, but I see no reason for the event from ~9:30-10:30 to also be half size.

Not sure if this is a bug or something I can change myself. Help would be appreciated!

**Comment by faiyaz-shaikh** [comment]
Fixed in 1.0.1 PR #135

**Comment by porum** [comment]
Now, if events are seamless (such as EventA's start time is same as EventB's end time), may be make they full width is better.

![B](https://user-images.githubusercontent.com/48167317/206683241-d12b31ff-0570-4209-8d23-4e80be971d35.jpg)
